apps/src/lib/node/ledger/ethereum_node/mod.rs

```diff
@@ -209,22 +208,19 @@ pub mod eth_fullnode {
         /// received from the Oracle process. If either, return the
         /// status.
         async fn wait(&mut self) -> Result<()> {
-            loop {
-                match self.process.try_wait() {
-                    Ok(Some(status)) => {
-                        return if status.success() {
-                            Ok(())
-                        } else {
-                            Err(Error::Runtime(status.to_string()))
-                        };
-                    }
-                    Ok(None) => {}
-                    Err(err) => return Err(Error::Runtime(err.to_string())),
+            use futures::future::{self, Either};
+
+            let child_proc = self.process.wait();
+            futures::pin_mut!(child_proc);
+
+            match future::select(&mut self.abort_recv, child_proc).await {
+                Either::Left((_abort_received, _)) => Err(Error::Oracle),
```

Comment:
```suggestion
                Either::Left(_) => Err(Error::Oracle),
```

Reply:
on second thoughts idk about this suggestion, maybe having `_abort_received` is more explicit about what is happening